sigar_windows.go

```diff
@@ -117,8 +118,70 @@ func (self *Cpu) populateFromPdh(values []uint64) {
 	self.Irq = values[3]
 }
 
+// Only get the first mount point for each volume
+func getVolumePathName(volumeName []uint16) (string, error) {
+	charsLen := C.DWORD(C.MAX_PATH)
+	var chars []uint16
+	for {
+		chars = make([]uint16, charsLen)
+		if C.GetVolumePathNamesForVolumeNameW((*C.WCHAR)(&volumeName[0]), (*C.WCHAR)(&chars[0]), C.MAX_PATH, &charsLen) == C.TRUE {
+			break
+		}
+		errno := syscall.GetLastError()
+		err, ok := errno.(syscall.Errno)
+		if !ok || err != C.ERROR_MORE_DATA {
+			return "", errno
+		}
+	}
```

Comment:
Yeah, looking at this with fresh eyes, it's super wrong. What we need to do is replace `C.MAX_PATH` in the call to `GetVolumePathNamesForVolumeNameW` with `len(chars)`. Then we'd start with a buffer of size `C.MAX_PATH` and grow it to `charsLen` if the first call fails. 

The convention on MSDN is to wrap these in loops (as above), but it seems a lot better to try twice and fail loudly instead of getting stuck in an infinite loop. I'll rewrite appropriately.